rust/libnewsboat/src/utils.rs

```diff
@@ -499,27 +501,49 @@ pub fn get_command_output(cmd: &str) -> String {
         .unwrap_or_else(|_| String::from(""))
 }
 
-// This function assumes that the user is not interested in command's output (not even errors on
-// stderr!), so it redirects everything to /dev/null.
+// This function assumes that the user is not interested in the command's output (not even errors
+// on stderr!), so it will close the spawned command's fds.
+// This used to be a simple std::process::Command::Spawn(), but this caused child processes to not
+// be reaped, hence the addition of double forking to this function.
+// Spawn() was replaced with a direct call to fork+execvp because it interacted badly with the
+// fork() call.
 pub fn run_command(cmd: &str, param: &str) {
-    let child = Command::new(cmd)
-        .arg(param)
-        // Prevent the command from blocking Newsboat by asking for input
-        .stdin(Stdio::null())
-        // Prevent the command from botching the screen by printing onto it.
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
-        .spawn();
-    if let Err(error) = child {
-        log!(
-            Level::Debug,
-            "utils::run_command: spawning a child for \"{}\" failed: {}",
-            cmd,
-            error
-        );
-    }
+    unsafe {
+        let mut forked_pid = fork();
+
+        match forked_pid {
+            -1 => {
+                // Parent process, fork failed.
+                log!(
+                    Level::Debug,
+                    "utils::run_command: failed to fork. Aborting run_command."
+                );
+                return;
+            }
+            0 => {
+                // Child process, continue and spawn the command.
+            }
+            _ => {
+                // Parent process, fork succeeded: reap child and return.
+                wait(&mut forked_pid);
```

Comment:
This waits for *any* child, not just `forked_pid`, because [the argument to `wait()` is not actually a PID](https://man7.org/linux/man-pages/man2/wait.2.html). I believe this can interfere with other threads that wait for a child to finish, "stealing" that info from them. It's probably safer to use `waitpid()` here.

Reply:
Sorry for wasting your time with this when I should just have read the manual 🤦.